### PR TITLE
Adyen Giving 2.0

### PR DESCRIPTION
**Pull request by zenit2001**
This PR contains the implementation of Adyen Giving 2.0

### CODE REVIEW

File: src/cartridges/int_adyen_overlay/cartridge/scripts/util/adyenHelper.js [hunk 1]
```diff
@@ -285,26 +283,10 @@ var adyenHelperObj = {
...
-      'directEbanking',
-      'vipps',
-      'sepadirectdebit',
-      'directdebit_GB',
-    ];
-    return availablePaymentMethods.indexOf(paymentMethod) !== -1;
+  // determines whether Adyen Giving is available based on the donation token 
+  isAdyenGivingAvailable(paymentInstrument) {
+    // Adyen giving is only available for BCMC in POS
+    return paymentInstrument.paymentTransaction.custom.Adyen_donationToken && paymentInstrument.paymentTransaction.custom.Adyen_paymentMethod !== 'bcmc' ? true : false;
```
Comment (zaiddreakh) [inline src/cartridges/int_adyen_overlay/cartridge/scripts/util/adyenHelper.js:289]:
```suggestion
    return paymentInstrument.paymentTransaction.custom.Adyen_donationToken && paymentInstrument.paymentTransaction.custom.Adyen_paymentMethod !== 'bcmc';
```

**Comment by amihajlovski** [comment]
Please update the PR title and labels :)

**Review by zaiddreakh — DISMISSED** [review]

**Review by amihajlovski — APPROVED** [review]

File: src/cartridges/int_adyen_overlay/cartridge/scripts/handleCustomObject.js [hunk 2]
```diff
@@ -71,7 +71,9 @@ function handle(customObj) {
 
   // split order ID by - and remove last split (which is the date)
   const orderIdSplit = customObj.custom.orderId.split('-').slice(0, -1);
```
Comment (zaiddreakh) [inline src/cartridges/int_adyen_overlay/cartridge/scripts/handleCustomObject.js:73]:
```suggestion
  const orderIdParts = customObj.custom.orderId.split('-');
  orderIdParts.pop();
```
slice creates a new array without the last element. pop uses same one

File: src/cartridges/int_adyen_overlay/cartridge/scripts/handleCustomObject.js [hunk 3]
```diff
@@ -71,7 +71,9 @@ function handle(customObj) {
...
-  const orderId = orderIdSplit.join('-');
+  // in case the splitted array contains more than 1 element (DONATION case), get only the last split (which is the order number)
+  const finalOrderIdSplit = orderIdSplit.length > 1 ? orderIdSplit.slice(-1) : orderIdSplit;
```
Comment (zaiddreakh) [inline src/cartridges/int_adyen_overlay/cartridge/scripts/handleCustomObject.js:75]:
```suggestion
  const relevantOrderIdParts = orderIdSplit.length > 1 ? orderIdSplit.slice(-1) : orderIdSplit; 
```

File: src/cartridges/int_adyen_overlay/cartridge/scripts/handleCustomObject.js [hunk 4]
```diff
@@ -71,7 +71,9 @@ function handle(customObj) {
 
   // split order ID by - and remove last split (which is the date)
   const orderIdSplit = customObj.custom.orderId.split('-').slice(0, -1);
-  const orderId = orderIdSplit.join('-');
+  // in case the splitted array contains more than 1 element (DONATION case), get only the last split (which is the order number)
+  const finalOrderIdSplit = orderIdSplit.length > 1 ? orderIdSplit.slice(-1) : orderIdSplit; 
+  const orderId = finalOrderIdSplit.join('-');
```
Comment (zaiddreakh) [inline src/cartridges/int_adyen_overlay/cartridge/scripts/handleCustomObject.js:76]:
```suggestion
  const orderId = relevantOrderIdParts.join('-');
```

**Review by zaiddreakh — APPROVED** [review]

**Review by amihajlovski — APPROVED** [review]

**Pull request merged**